lib/screens/v2/candidate_generator/dup/departures.ex

```diff
@@ -501,6 +487,52 @@ defmodule Screens.V2.CandidateGenerator.Dup.Departures do
 
   defp get_overnight_schedules_for_section(_, _, _, _, _, _, _), do: []
 
+  defp get_overnight_departure_for_route(
+         nil,
+         _first_schedule_tomorrow,
```

Comment:
Does this fully replicate the first condition in the original original logic? What if `last_schedule_today` is not nil but `first_schedule_tomorrow` is?

Reply:
It replicates half of it. We know for sure that `last_schedule_today == nil` means something went wrong so don't show overnight just in case. If `first_schedule_tomorrow == nil`, That may be because there are no schedules for that route/direction the following day. In that case, we show a row for the route/direction with a moon icon instead of a time (line 514 is where this is handled).